identity-iota/src/did/doc/iota_document.rs

```diff
@@ -380,6 +385,49 @@ impl IotaDocument {
     self.document.insert_method(method.into(), scope)
   }
 
+  // Attaches the given relationship to the given method, if the method exists.
+  pub fn attach_method_relationship(&mut self, did_url: IotaDIDUrl, relationship: MethodRelationship) -> Result<bool> {
+    // Ensure the method exists
+    self
+      .document
+      .resolve_method(did_url.url())
+      .ok_or(identity_did::Error::QueryMethodNotFound)?;
+
+    let core_did_url: CoreDIDUrl = CoreDIDUrl::from(did_url);
+    let method_ref = MethodRef::Refer(core_did_url);
+
+    let result = match relationship {
+      MethodRelationship::Authentication => self.document.authentication_mut().append(method_ref),
+      MethodRelationship::AssertionMethod => self.document.assertion_method_mut().append(method_ref),
+      MethodRelationship::KeyAgreement => self.document.key_agreement_mut().append(method_ref),
+      MethodRelationship::CapabilityDelegation => self.document.capability_delegation_mut().append(method_ref),
+      MethodRelationship::CapabilityInvocation => self.document.capability_invocation_mut().append(method_ref),
+    };
+
+    Ok(result)
+  }
+
+  // Detaches the given relationship from the given method, if the method exists.
+  pub fn detach_method_relationship(&mut self, did_url: IotaDIDUrl, relationship: MethodRelationship) -> Result<()> {
+    // Ensure the method exists
+    self
+      .document
+      .resolve_method(did_url.url())
+      .ok_or(identity_did::Error::QueryMethodNotFound)?;
+
+    let core_did_url: CoreDIDUrl = CoreDIDUrl::from(did_url);
```

Comment:
The query is a good suggestion, makes it more generic. Unfortunately, we have to introduce a clone, because
of `cannot borrow *self as mutable because it is also borrowed as immutable` (mutably borrowed when mutating the relationships. But I changed it to use the query.